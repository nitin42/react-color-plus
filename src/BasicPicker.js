import React from 'react'
import { css, injectGlobal } from 'emotion'
import { ColorExtractor } from 'react-color-extractor'
import generateColors from 'randomcolor'

import ColorInput from './components/ColorInputField'
import ColorBlock from './components/ColorBlock'
import ActiveColor from './components/ActiveColor'
import Container from './components/Container'
import Swatches from './components/SwatchesContainer'
import Color from './components/Color'
import ImagePicker from './components/ImagePicker'

injectGlobal`body { background: mistyrose; }`

export class BasicPicker extends React.Component {
	state = {
		color: '#088da5',
		swatches: ['#5a80b4', '#40e0d0', '#088da5', '#f6546a', '#cac8a0', '#0079cf'],
		image: null,
	}

	componentDidMount() {
		const uploadElement = document.getElementById('uploader')
		const imageIcon = document.getElementById('image-icon')

		imageIcon.addEventListener('click', e => {
			if (uploadElement) {
				uploadElement.click()
			}

			e.preventDefault()
		})

		document.addEventListener('keydown', e => {
			if (e.which === 8) {
				this.setState({ image: null })
			}
		})
	}

	generate = e => {
		let i = 0
		let newColors = new Set()

		while (i < 10) {
			newColors.add(generateColors())
			i++
		}

		this.setState({ swatches: [...newColors] })
	}

	uploadFiles = e => {
		this.setState({ image: window.URL.createObjectURL(e.target.files[0]) })
	}

	onChange = color => this.setState({ color })

	renderSwatches = () => {
		const { swatches } = this.state

		return swatches.map((swatch, i) => (
			<span key={i}>
				<Color value={swatch} updateColor={e => this.updateSwatch(swatch)} />
			</span>
		))
	}

	updateSwatch = color => this.setState({ color })

	render() {
		return (
			<div style={{ display: 'flex', justifyContent: 'center', marginTop: 30 }}>
				<Container>
					{this.state.image === null ? (
						<ColorBlock color={this.state.color}>
							<ActiveColor color={this.state.color} />
						</ColorBlock>
					) : (
						<ColorExtractor getColors={colors => this.setState({ swatches: [...colors] })}>
							<img
								src={this.state.image}
								className={css`
									height: 110px;
									border-radius: 6px 6px 0px 0px;
									display: flex;
									align-items: center;
									justify-content: center;
									position: relative;
									width: 100%;
								`}
							/>
						</ColorExtractor>
					)}
					<div style={{ padding: 10 }}>
						<Swatches>
							{this.renderSwatches()}
							<div style={{ clear: 'both' }} />
						</Swatches>
						<ColorInput value={this.state.color} onChange={this.onChange} />
						<div
							className={css`
								margin-top: 10px;
								display: flex;
								justify-content: center;
								cursor: pointer;
							`}
						>
							<span>
								<input
									id="uploader"
									style={{ display: 'none' }}
									type="file"
									accept="image/*"
									onChange={this.uploadFiles}
								/>
								<i id="image-icon" className="fas fa-image" />
							</span>
							<span style={{ marginLeft: 10 }} onClick={this.generate}>
								<i id="image-icon" className="fas fa-palette" />
							</span>
						</div>
					</div>
				</Container>
			</div>
		)
	}
}
