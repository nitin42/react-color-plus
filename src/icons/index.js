import React from 'react'
import styled from 'react-emotion'

const StyledSVG = styled('svg')`
  display: inline-block;
  width: 20px;
  position: relative;
  top: 8px;
  left: -5px;
`

const ToolsSVG = styled('svg')`
  cursor: pointer;
`

export const createStyledSVG = ({
  height,
  width,
  viewBox,
  color,
  title,
  d
}) => (
  <StyledSVG width={width} height={height} viewBox={viewBox}>
    <title>{title}</title>
    <path fill={color} d={d} />
  </StyledSVG>
)

export const SaturatorIcon = ({ width, height, color }) =>
  createStyledSVG({
    height,
    width,
    color,
    viewBox: '0 0 24 24',
    d:
      'M12 15.422l3.75 2.25-0.984-4.266 3.328-2.906-4.406-0.375-1.688-4.031v9.328zM21.984 9.234l-5.438 4.734 1.641 7.031-6.188-3.75-6.188 3.75 1.641-7.031-5.438-4.734 7.172-0.609 2.813-6.609 2.813 6.609z'
  })

export const DesaturatorIcon = ({ width, height, color }) =>
  createStyledSVG({
    height,
    width,
    color,
    viewBox: '0 0 26 28',
    d:
      'M23.859 22.625c1.172 1.859 0.344 3.375-1.859 3.375h-18c-2.203 0-3.031-1.516-1.859-3.375l7.859-12.391v-6.234h-1c-0.547 0-1-0.453-1-1s0.453-1 1-1h8c0.547 0 1 0.453 1 1s-0.453 1-1 1h-1v6.234zM11.688 11.297l-4.25 6.703h11.125l-4.25-6.703-0.313-0.484v-6.813h-2v6.813z'
  })

export const ColorSpinIcon = ({ width, height, color }) =>
  createStyledSVG({
    height,
    width,
    color,
    viewBox: '0 0 24 28',
    d:
      'M23.609 16.5c0 0.031 0 0.078-0.016 0.109-1.328 5.531-5.891 9.391-11.656 9.391-3.047 0-6-1.203-8.219-3.313l-2.016 2.016c-0.187 0.187-0.438 0.297-0.703 0.297-0.547 0-1-0.453-1-1v-7c0-0.547 0.453-1 1-1h7c0.547 0 1 0.453 1 1 0 0.266-0.109 0.516-0.297 0.703l-2.141 2.141c1.469 1.375 3.422 2.156 5.437 2.156 2.781 0 5.359-1.437 6.813-3.813 0.375-0.609 0.562-1.203 0.828-1.828 0.078-0.219 0.234-0.359 0.469-0.359h3c0.281 0 0.5 0.234 0.5 0.5zM24 4v7c0 0.547-0.453 1-1 1h-7c-0.547 0-1-0.453-1-1 0-0.266 0.109-0.516 0.297-0.703l2.156-2.156c-1.484-1.375-3.437-2.141-5.453-2.141-2.781 0-5.359 1.437-6.813 3.813-0.375 0.609-0.562 1.203-0.828 1.828-0.078 0.219-0.234 0.359-0.469 0.359h-3.109c-0.281 0-0.5-0.234-0.5-0.5v-0.109c1.344-5.547 5.953-9.391 11.719-9.391 3.063 0 6.047 1.219 8.266 3.313l2.031-2.016c0.187-0.187 0.438-0.297 0.703-0.297 0.547 0 1 0.453 1 1z'
  })

export const ColorDarkenerIcon = ({ width, height, color }) => (
  <StyledSVG width={width} height={height} viewBox="0 0 32 32">
    <title>Color Darkener</title>
    <path
      fill={color}
      d="M24.633 22.184c-8.188 0-14.82-6.637-14.82-14.82 0-2.695 0.773-5.188 2.031-7.363-6.824 1.968-11.844 8.187-11.844 15.644 0 9.031 7.32 16.355 16.352 16.355 7.457 0 13.68-5.023 15.648-11.844-2.18 1.254-4.672 2.028-7.367 2.028z"
    />
  </StyledSVG>
)

export const ColorBrightenerIcon = ({ width, height, color }) => (
  <StyledSVG width={width} height={height} viewBox="0 0 32 32">
    <title>Color Brightener</title>
    <path
      fill={color}
      d="M16 9c-3.859 0-7 3.141-7 7s3.141 7 7 7 7-3.141 7-7c0-3.859-3.141-7-7-7zM16 21c-2.762 0-5-2.238-5-5s2.238-5 5-5 5 2.238 5 5-2.238 5-5 5zM16 7c0.552 0 1-0.448 1-1v-2c0-0.552-0.448-1-1-1s-1 0.448-1 1v2c0 0.552 0.448 1 1 1zM16 25c-0.552 0-1 0.448-1 1v2c0 0.552 0.448 1 1 1s1-0.448 1-1v-2c0-0.552-0.448-1-1-1zM23.777 9.635l1.414-1.414c0.391-0.391 0.391-1.023 0-1.414s-1.023-0.391-1.414 0l-1.414 1.414c-0.391 0.391-0.391 1.023 0 1.414s1.023 0.391 1.414 0zM8.223 22.365l-1.414 1.414c-0.391 0.391-0.391 1.023 0 1.414s1.023 0.391 1.414 0l1.414-1.414c0.391-0.392 0.391-1.023 0-1.414s-1.023-0.392-1.414 0zM7 16c0-0.552-0.448-1-1-1h-2c-0.552 0-1 0.448-1 1s0.448 1 1 1h2c0.552 0 1-0.448 1-1zM28 15h-2c-0.552 0-1 0.448-1 1s0.448 1 1 1h2c0.552 0 1-0.448 1-1s-0.448-1-1-1zM8.221 9.635c0.391 0.391 1.024 0.391 1.414 0s0.391-1.023 0-1.414l-1.414-1.414c-0.391-0.391-1.023-0.391-1.414 0s-0.391 1.023 0 1.414l1.414 1.414zM23.779 22.363c-0.392-0.391-1.023-0.391-1.414 0s-0.392 1.023 0 1.414l1.414 1.414c0.391 0.391 1.023 0.391 1.414 0s0.391-1.023 0-1.414l-1.414-1.414z"
    />
  </StyledSVG>
)

export const ImagePickerIcon = ({ width, height, color, ...rest }) => (
  <ToolsSVG {...rest} width={width} height={height} viewBox="0 0 30 28">
    <title>Image picker</title>
    <path
      fill={color}
      d="M10 9c0 1.656-1.344 3-3 3s-3-1.344-3-3 1.344-3 3-3 3 1.344 3 3zM26 15v7h-22v-3l5-5 2.5 2.5 8-8zM27.5 4h-25c-0.266 0-0.5 0.234-0.5 0.5v19c0 0.266 0.234 0.5 0.5 0.5h25c0.266 0 0.5-0.234 0.5-0.5v-19c0-0.266-0.234-0.5-0.5-0.5zM30 4.5v19c0 1.375-1.125 2.5-2.5 2.5h-25c-1.375 0-2.5-1.125-2.5-2.5v-19c0-1.375 1.125-2.5 2.5-2.5h25c1.375 0 2.5 1.125 2.5 2.5z"
    />
  </ToolsSVG>
)

export const ShadesGeneratorIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 24 28">
    <title>Shades</title>
    <path
      fill={color}
      d="M12 22.5v-17c-4.688 0-8.5 3.813-8.5 8.5s3.813 8.5 8.5 8.5zM24 14c0 6.625-5.375 12-12 12s-12-5.375-12-12 5.375-12 12-12 12 5.375 12 12z"
    />
  </ToolsSVG>
)

export const TintsGeneratorIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 16 28">
    <title>Tints</title>
    <path
      fill={color}
      d="M8 18c0-0.391-0.125-0.766-0.313-1.078-0.203-0.313-1.031-1.375-1.359-2.422-0.047-0.172-0.203-0.25-0.328-0.25s-0.281 0.078-0.328 0.25c-0.328 1.047-1.156 2.109-1.359 2.422-0.187 0.313-0.313 0.688-0.313 1.078 0 1.109 0.891 2 2 2s2-0.891 2-2zM16 16c0 4.422-3.578 8-8 8s-8-3.578-8-8c0-1.578 0.484-3.047 1.266-4.297 0.797-1.25 4.141-5.484 5.406-9.703 0.203-0.672 0.828-1 1.328-1s1.141 0.328 1.328 1c1.266 4.219 4.609 8.453 5.406 9.703s1.266 2.719 1.266 4.297z"
    />
  </ToolsSVG>
)

export const ClipboardIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 20 20">
    <title>Clipboard</title>
    <path
      fill={color}
      d="M15.6 2l-1.2 3h-8.8l-1.2-3c-0.771 0-1.4 0.629-1.4 1.4v15.2c0 0.77 0.629 1.4 1.399 1.4h11.2c0.77 0 1.4-0.631 1.4-1.4v-15.2c0.001-0.771-0.63-1.4-1.399-1.4zM13.6 4l0.9-2h-2.181l-0.719-2h-3.2l-0.72 2h-2.18l0.899 2h7.201z"
    />
  </ToolsSVG>
)

export const SwatchesGeneratorIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 24 24">
    <title>Swatches Generator</title>
    <path
      fill={color}
      d="M17.484 12c0.844 0 1.5-0.656 1.5-1.5s-0.656-1.5-1.5-1.5-1.5 0.656-1.5 1.5 0.656 1.5 1.5 1.5zM14.484 8.016c0.844 0 1.5-0.656 1.5-1.5s-0.656-1.5-1.5-1.5-1.5 0.656-1.5 1.5 0.656 1.5 1.5 1.5zM9.516 8.016c0.844 0 1.5-0.656 1.5-1.5s-0.656-1.5-1.5-1.5-1.5 0.656-1.5 1.5 0.656 1.5 1.5 1.5zM6.516 12c0.844 0 1.5-0.656 1.5-1.5s-0.656-1.5-1.5-1.5-1.5 0.656-1.5 1.5 0.656 1.5 1.5 1.5zM12 3c4.969 0 9 3.609 9 8.016 0 2.766-2.25 4.969-5.016 4.969h-1.734c-0.844 0-1.5 0.656-1.5 1.5 0 0.375 0.141 0.703 0.375 0.984s0.375 0.656 0.375 1.031c0 0.844-0.656 1.5-1.5 1.5-4.969 0-9-4.031-9-9s4.031-9 9-9z"
    />
  </ToolsSVG>
)

export const ResetIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 32 32">
    <title>Reset state</title>
    <path
      fill={color}
      d="M32 16c0-8.836-7.164-16-16-16-8.837 0-16 7.164-16 16 0 8.837 7.163 16 16 16 8.836 0 16-7.163 16-16zM8 16l8-8v6h8v4h-8v6l-8-8z"
    />
  </ToolsSVG>
)

export const GenerateGradientIcon = ({ width, height, color }) => (
  <ToolsSVG width={width} height={height} viewBox="0 0 32 32">
    <title>Gradient generator</title>
    <path
      fill={color}
      d="M32 12h-12l4.485-4.485c-2.267-2.266-5.28-3.515-8.485-3.515s-6.219 1.248-8.485 3.515c-2.266 2.267-3.515 5.28-3.515 8.485s1.248 6.219 3.515 8.485c2.267 2.266 5.28 3.515 8.485 3.515s6.219-1.248 8.485-3.515c0.189-0.189 0.371-0.384 0.546-0.583l3.010 2.634c-2.933 3.349-7.239 5.464-12.041 5.464-8.837 0-16-7.163-16-16s7.163-16 16-16c4.418 0 8.418 1.791 11.313 4.687l4.687-4.687v12z"
    />
  </ToolsSVG>
)
