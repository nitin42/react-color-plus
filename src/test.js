import React from 'react'
import ReactDOM from 'react-dom'
import { css } from 'emotion'

import { BasicPicker, GradientPicker } from './index'

class App extends React.Component {
  state = {
    color: 'hotpink',
    gradient: ''
  }

  render() {
    return (
      <div
        className={css`
          display: flex;
          justify-content: center;
          align-items: center;
          flex-direction: column;
          margin-top: 30px;
        `}
      >
        <GradientPicker
          theme="light"
          mode="linear"
          direction="to top"
          reverse
          getGradient={gradient => this.setState({ gradient })}
        />
        <h1
          className={css`
            background-image: ${this.state.gradient};
            -webkit-background-clip: text;
            -webkit-text-fill-color: transparent;
          `}
        >
          React Color Tools
        </h1>
      </div>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('root'))
