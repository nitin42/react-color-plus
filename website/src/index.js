import React from 'react'
import { render } from 'react-dom'
import { css, injectGlobal } from 'emotion'
import { BasicPicker, SchemePicker, GradientPicker } from 'react-color-tools'
import coolhue from 'coolhue'

const BASIC_PICKER = 'Basic Color Picker'
const SCHEME_PICKER = 'Color Scheme Picker'
const GRADIENT_PICKER = 'Gradient Picker'

injectGlobal`
body {
  font-family: 'Source Sans Pro', sans-serif;
  color: #2f2f2f;
}
`

const shadow = css`
  -webkit-box-shadow: 10px 10px 9px -11px rgba(122, 121, 122, 1);
  -moz-box-shadow: 10px 10px 9px -11px rgba(122, 121, 122, 1);
  box-shadow: 10px 10px 9px -11px rgba(122, 121, 122, 1);
`

const Heading = props => (
  <h1
    className={css`
      font-size: 4em;
      ${props.gradientPicker
        ? `background-image: ${props.color};
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
        `
        : `color: ${props.color};
        `};
      @media only screen and (max-width: 480px) {
        font-size: 2.3em;
      }
    `}
  >
    {props.children}
  </h1>
)

Heading.defaultProps = {
  gradientPicker: false
}

const Description = props => (
  <p
    className={css`
      font-family: 'Satisfy', cursive;
      font-size: 1.5em;
      margin-top: -20px;
      color: #4f4f4f;
      @media only screen and (max-width: 480px) {
        font-size: 0.85em;
      }
    `}
  >
    A set of tools as React components for working with{' '}
    <span
      onClick={props.updateGradient}
      className={css`
        background-image: ${props.gradient};
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
        user-select: none;
        cursor: pointer;
      `}
    >
      colors
    </span>
  </p>
)

const GitHubLink = props => (
  <div style={{ marginTop: 20 }}>
    <a
      className="github-button"
      href="https://github.com/nitin42/react-color-tools"
      data-size="large"
      data-show-count="true"
      aria-label="Star nitin42/react-color-tools on GitHub"
    >
      Star
    </a>
  </div>
)

const Container = props => (
  <div
    className={css`
      display: flex;
      justify-content: center;
      align-items: center;
      margin-top: 20px;
      flex-direction: ${props.direction};
    `}
  >
    {props.children}
  </div>
)

const Link = ({ underline, url, children, ...rest }) => (
  <a
    className={css`
      font-size: 1.5em;
      text-decoration: none;
      color: #4f4f4f;
      border-bottom: 2px solid #4f4f4f;
      @media only screen and (max-width: 480px) {
        font-size: 1.2em;
      }
    `}
    href={url}
    target="_blank"
    rel="noopener noreferrer"
    {...rest}
  >
    {children}
  </a>
)

const Footer = props => (
  <footer style={{ marginTop: 5, bottom: 0 }}>
    <p
      style={{
        fontSize: 18,
        color: '#4f4f4f'
      }}
    >
      Made with{' '}
      <span role="img" aria-label="heart">
        ❤️
      </span>
      ️ by{' '}
      <Link url="https://nitin-tulswani.surge.sh/" style={{ fontSize: 18 }}>
        Nitin Tulswani
      </Link>
    </p>
  </footer>
)

class App extends React.Component {
  state = {
    color: 'pink',
    pickers: ['Basic Color Picker', 'Gradient Picker', 'Color Scheme Picker'],
    currentPicker: 'Basic Color Picker',
    gradient: '',
    descGrad: coolhue.getGradientStyle(5)
  }

  changeFormat = e => this.setState({ currentPicker: e.target.value })

  renderPickerOptions = () => {
    const { pickers } = this.state

    return pickers.map(picker => (
      <option value={picker} key={picker}>
        {picker}
      </option>
    ))
  }

  getRandomValue = (min, max) => Math.random() * (max - min) + min

  updateGradient = e => {
    const number = Math.floor(this.getRandomValue(1, 60))

    this.setState({ descGrad: coolhue.getGradientStyle(number) })
  }

  render() {
    return (
      <div>
        <Container direction={'column'}>
          <Heading
            gradientPicker={this.state.currentPicker === GRADIENT_PICKER}
            color={
              this.state.currentPicker !== GRADIENT_PICKER
                ? this.state.color
                : this.state.gradient
            }
          >
            React Color Tools
          </Heading>
          <Description
            gradient={this.state.descGrad}
            updateGradient={this.updateGradient}
          />
          <div className={shadow}>
            {this.state.currentPicker === BASIC_PICKER && (
              <BasicPicker
                triangle={false}
                color={this.state.color}
                onChange={color => this.setState({ color })}
                showTools
              />
            )}
            {this.state.currentPicker === SCHEME_PICKER && (
              <SchemePicker
                scheme="analogous"
                color={this.state.color}
                onChange={color => this.setState({ color })}
              />
            )}
            {this.state.currentPicker === GRADIENT_PICKER && (
              <GradientPicker
                getGradient={gradient => this.setState({ gradient })}
              />
            )}
          </div>
          <div style={{ marginTop: 20 }}>
            <select onChange={this.changeFormat}>
              {this.renderPickerOptions()}
            </select>
          </div>
          <GitHubLink />
          <div style={{ marginTop: 20, marginBottom: 10 }}>
            <Link url="https://github.com/nitin42/react-color-tools/tree/master/docs">
              Read the detailed documentation
            </Link>
          </div>
          <Footer />
        </Container>
      </div>
    )
  }
}

render(<App />, document.getElementById('root'))
