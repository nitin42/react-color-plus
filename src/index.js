/* eslint-disable */
import React from 'react'
import ReactDOM from 'react-dom'

import { css } from 'emotion'
import BasicPicker from './pickers/BasicPicker'
import GradientPicker from './pickers/GradientPicker'
import SchemePicker from './pickers/SchemePicker'

const DEFAULT_COLOR_ONE = '#81FFEF'
const DEFAULT_COLOR_TWO = '#F067B4'

const PickerBasic = props => <BasicPicker {...props} />

const PickerGradient = props => <GradientPicker {...props} />

const directions = {
  one: 'farthest-corner ellipse at top left',
  two: 'ellipse at center',
  three: 'at top-left',
  four: 'at right'
}

/* eslint-disable */
class App extends React.Component {
  state = {
    color: 'hotpink',
    gradient: ''
  }

  render() {
    return (
      <div>
        <SchemePicker
          color={this.state.color}
          onChange={color => {
            this.setState({ color })
          }}
        />
        <h1 style={{ color: this.state.color }}>SCHEME PICKER</h1>
      </div>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('root'))
