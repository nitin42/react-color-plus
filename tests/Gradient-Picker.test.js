import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'

import { GradientPicker } from '../src'

class App extends React.Component {
  state = { gradient: '' }

  render() {
    const { gradient: grad } = this.state

    return (
      <div>
        <GradientPicker
          mode={this.props.mode}
          direction={this.props.direction}
          getGradient={gradient => this.setState({ gradient })}
        />
        <h1
          style={{
            backgroundImage: grad,
            WebkitBackgroundClip: 'text',
            WebkitTextFillColor: 'transparent'
          }}
        >
          React Gradient Picker
        </h1>
      </div>
    )
  }
}

describe('Test GradientPicker API', () => {
  it('should render the gradient picker', () => {
    const tree = renderer.create(<GradientPicker />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the gradient picker with dark theme', () => {
    const tree = renderer.create(<GradientPicker theme="dark" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should update the state of parent component', () => {
    const Wrapper = mount(<App />)

    expect(Wrapper.state('gradient')).toEqual(
      'linear-gradient(to top, rgb(129, 255, 239) 0%, rgb(240, 103, 180) 100%)'
    )
  })

  it('should create gradient with mode and direction prop', () => {
    const Wrapper = mount(<App mode="linear" direction="to left" />)
    expect(Wrapper.state('gradient')).toEqual(
      'linear-gradient(to left, rgb(129, 255, 239) 0%, rgb(240, 103, 180) 100%)'
    )
  })

  it('should create new gradient based when color input changes', () => {
    const Wrapper = mount(<App />)

    expect(Wrapper.state('gradient')).toEqual(
      'linear-gradient(to top, rgb(129, 255, 239) 0%, rgb(240, 103, 180) 100%)'
    )

    Wrapper.find('ColorInputField').forEach((node, i) => {
      if (i === 1) {
        node.simulate('change', { target: { value: 'red' } })
      } else {
        node.simulate('change', { target: { value: 'orange' } })
      }
    })

    expect(Wrapper.state('gradient')).toEqual(
      'linear-gradient(to top, rgb(255, 165, 0) 0%, rgb(255, 0, 0) 100%)'
    )
  })

  it('should update the gradient when color stop position changes', () => {
    const Wrapper = mount(<App />)

    Wrapper.find('Slider').forEach((node, i) => {
      if (i === 1) {
        node.simulate('change', { target: { value: '8' } })
      } else {
        node.simulate('change', { target: { value: '4' } })
      }
    })

    expect(Wrapper.state('gradient')).toEqual(
      'linear-gradient(to top, rgb(129, 255, 239) 0%, rgb(129, 255, 239) 20%, rgb(240, 103, 180) 80%, rgb(240, 103, 180) 100%)'
    )
  })
})
