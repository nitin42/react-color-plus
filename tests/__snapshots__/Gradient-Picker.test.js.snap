// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GradientPicker API should render the gradient picker 1`] = `
<div
  className="css-dkpzwh"
>
  <div
    className="css-3ef40x"
    style={
      Object {
        "backgroundImage": "linear-gradient(to right, rgb(129, 255, 239) 0%, rgb(240, 103, 180) 100%)",
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <div>
      <input
        className="css-sskoaq"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        spellCheck="false"
        value="#81FFEF"
      />
      <span>
        <label
          className="label css-nj1yzf"
          color="#1f1f1f"
        >
          Stops
        </label>
        <input
          className="css-1705gpz"
          max="10"
          min="0"
          onChange={[Function]}
          step="0.01"
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
          type="range"
          value={0}
        />
      </span>
    </div>
    <div>
      <input
        className="css-sskoaq"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        spellCheck="false"
        value="#F067B4"
      />
      <span>
        <label
          className="label css-nj1yzf"
          color="#1f1f1f"
        >
          Stops
        </label>
        <input
          className="css-1705gpz"
          max="10"
          min="0"
          onChange={[Function]}
          step="0.01"
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
          type="range"
          value={0}
        />
      </span>
    </div>
    <div
      className="css-upj78"
    >
      <span
        aria-label="Copied"
        className="no-tooltip css-1v5ga19"
        id="gradient-clipboard"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          className="css-1h5x3dy"
          height={18}
          viewBox="0 0 20 20"
          width={19}
        >
          <title>
            clipboard
          </title>
          <path
            d="M15.6 2l-1.2 3h-8.8l-1.2-3c-0.771 0-1.4 0.629-1.4 1.4v15.2c0 0.77 0.629 1.4 1.399 1.4h11.2c0.77 0 1.4-0.631 1.4-1.4v-15.2c0.001-0.771-0.63-1.4-1.399-1.4zM13.6 4l0.9-2h-2.181l-0.719-2h-3.2l-0.72 2h-2.18l0.899 2h7.201z"
            fill="#1f1f1f"
          />
        </svg>
      </span>
      <div
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <span
          className="css-1v5ga19"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
          title="Gradient generator"
        >
          <svg
            className="css-1h5x3dy"
            height={20}
            viewBox="0 0 32 32"
            width={17.5}
          >
            <title>
              gradient generator
            </title>
            <path
              d="M32 12h-12l4.485-4.485c-2.267-2.266-5.28-3.515-8.485-3.515s-6.219 1.248-8.485 3.515c-2.266 2.267-3.515 5.28-3.515 8.485s1.248 6.219 3.515 8.485c2.267 2.266 5.28 3.515 8.485 3.515s6.219-1.248 8.485-3.515c0.189-0.189 0.371-0.384 0.546-0.583l3.010 2.634c-2.933 3.349-7.239 5.464-12.041 5.464-8.837 0-16-7.163-16-16s7.163-16 16-16c4.418 0 8.418 1.791 11.313 4.687l4.687-4.687v12z"
              fill="#1f1f1f"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test GradientPicker API should render the gradient picker with dark theme 1`] = `
<div
  className="css-1v1bp1x"
>
  <div
    className="css-3ef40x"
    style={
      Object {
        "backgroundImage": "linear-gradient(to right, rgb(129, 255, 239) 0%, rgb(240, 103, 180) 100%)",
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <div>
      <input
        className="css-sskoaq"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        spellCheck="false"
        value="#81FFEF"
      />
      <span>
        <label
          className="label css-njkhub"
          color="rgb(255, 255, 255)"
        >
          Stops
        </label>
        <input
          className="css-1lpha99"
          max="10"
          min="0"
          onChange={[Function]}
          step="0.01"
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
          type="range"
          value={0}
        />
      </span>
    </div>
    <div>
      <input
        className="css-sskoaq"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        spellCheck="false"
        value="#F067B4"
      />
      <span>
        <label
          className="label css-njkhub"
          color="rgb(255, 255, 255)"
        >
          Stops
        </label>
        <input
          className="css-1lpha99"
          max="10"
          min="0"
          onChange={[Function]}
          step="0.01"
          style={
            Object {
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
          type="range"
          value={0}
        />
      </span>
    </div>
    <div
      className="css-upj78"
    >
      <span
        aria-label="Copied"
        className="no-tooltip css-1v5ga19"
        id="gradient-clipboard"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <svg
          className="css-1h5x3dy"
          height={18}
          viewBox="0 0 20 20"
          width={19}
        >
          <title>
            clipboard
          </title>
          <path
            d="M15.6 2l-1.2 3h-8.8l-1.2-3c-0.771 0-1.4 0.629-1.4 1.4v15.2c0 0.77 0.629 1.4 1.399 1.4h11.2c0.77 0 1.4-0.631 1.4-1.4v-15.2c0.001-0.771-0.63-1.4-1.399-1.4zM13.6 4l0.9-2h-2.181l-0.719-2h-3.2l-0.72 2h-2.18l0.899 2h7.201z"
            fill="rgb(255, 255, 255)"
          />
        </svg>
      </span>
      <div
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <span
          className="css-1v5ga19"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
          title="Gradient generator"
        >
          <svg
            className="css-1h5x3dy"
            height={20}
            viewBox="0 0 32 32"
            width={17.5}
          >
            <title>
              gradient generator
            </title>
            <path
              d="M32 12h-12l4.485-4.485c-2.267-2.266-5.28-3.515-8.485-3.515s-6.219 1.248-8.485 3.515c-2.266 2.267-3.515 5.28-3.515 8.485s1.248 6.219 3.515 8.485c2.267 2.266 5.28 3.515 8.485 3.515s6.219-1.248 8.485-3.515c0.189-0.189 0.371-0.384 0.546-0.583l3.010 2.634c-2.933 3.349-7.239 5.464-12.041 5.464-8.837 0-16-7.163-16-16s7.163-16 16-16c4.418 0 8.418 1.791 11.313 4.687l4.687-4.687v12z"
              fill="rgb(255, 255, 255)"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
